lib/foreman_templates/engine.rb

```diff
@@ -16,6 +16,17 @@ class Engine < ::Rails::Engine
     initializer 'foreman_templates.register_plugin', :before => :finisher_hook do
       Foreman::Plugin.register :foreman_templates do
         requires_foreman '>= 1.12'
+
+        security_block :templates do
+          permission :import_templates, {
+            :template => [:import],
+            :"api/v2/template" => [:import]
+          }, :resource_type => 'Template' 
+        end
+
+        role = Role.where(:name => 'Manager').first
+        role.add_permissions!('import_templates') unless role.permissions.find_by_name('import_templates')
```

Comment:
So what happens is that this code get executed whenever the rails environment is loaded which happens also during migration. Since the way we run tests we do the plugin migrations before the DB of foreman core is seeded, the role does not exist yet so it fails on `role` variable being nil.

You can wrap the code in if condition to skip adding permission if the role does not exist. Note that the same code will be executed later when db is seeded or rails server starts so the permission will be added later when DB is ready.

```ruby
role = Role.where(:name => 'Manager').first
if role
  role.add_permissions!('import_templates') unless role.permissions.find_by_name('import_templates')
end
```

adding a explaining comment there would be a good idea. Once https://github.com/theforeman/foreman/pull/4168 is resolved we could update this code accordingly.